contracts/Sale.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity ^0.8.0;
+
+import "./sale/SaleBase.sol";
+import "./Freeport.sol";
+
+contract Sale is SaleBase {
+
+  	function initialize(Freeport freeport, address stablecoin) public initializer {
+  	  __SaleBase_init(freeport, stablecoin);
+  	}
+
+    /**
+     * Stores nft id and its quantity for sale. 
+     */
+    struct Nft {
+        uint256 price;
+        uint256 quantity;
+    }
+
+  	/**
+  	 * Mapping represents relation such as 
+     * seller address => (nft ID => structure containing price, quantity) 
+  	 */
+  	mapping(address => mapping(uint => Nft)) nftPrice;
```

Comment:
Is this feature an upgrade for Davinci or for the new major release of Freeport?

If the former, then what happens to the existing offers?

Reply:
will be a release if the new bid structure. past bids will be lost
though I can make it back compatible by storing quantity in another mapping. Which should I go with?

Reply:
It's actually new deployment. Don't think we can link it with freeport storage

Reply:
Ok, so this PR must go into the next major release (incompatible).